src/db/util/conditions.ts

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable @typescript-eslint/no-namespace */
+// we use namespaces here to neatly organize and scope the many type and type guard definitions
+
+import Knex = require('knex');
+
+/**
+ * Symbols to use for where condition construction
+ */
+export namespace ConditionSymbols {
+  export const BUILDER = Symbol('builder');
+  export const AND = Symbol('and');
+  export const OR = Symbol('or');
+
+  /**
+   * Symbols to use for where condition construction
+   */
+  export const Cond = {
+    BUILDER: BUILDER,
+    AND: AND,
+    OR: OR,
+  } as const;
+}
+
+export const Cond = ConditionSymbols.Cond;
+
+/**
+ * Symbols to use when constructing conditions for a single property
+ */
+export namespace PropertySymbols {
+  export const IN = Symbol('in');
+  export const NOT_IN = Symbol('not in');
+
+  /**
+   * Symbols to use when constructing conditions for a single property
+   */
+  export const Op = {
+    IN: IN,
+    NOT_IN: NOT_IN,
```

Comment:
Great stuff! I would also like us to be able to do `IS NOT NULL` check and `LIKE` without using a builder. At least those two, an maybe `LT`, `LTE`, `GT` and `GTE` should be added, mainly for `date` type comparison.

Reply:
I was thinking about that too, unsure whether it was worth adding before we actually had a use, but it won't be too much extra work :)